lib/wallet/wallet.js

```diff
@@ -832,6 +882,28 @@ class Wallet extends EventEmitter {
     return key;
   }
 
+  /**
+   * Derive an address (doesn't increments depth) without a lock.
+   * @private
+   * @param {(Number|String)?} acct
+   * @param {Number} branch
+   * @param {Number} index
+   * @returns {Promise} - Returns {@link WalletKey}.
+   */
+
+  async _deriveKey(acct, branch, index) {
+    const account = await this.getAccount(acct);
+
+    if (!account)
+      throw new Error('Account not found.');
+
+    const b = this.db.batch();
```

Comment:
There is no need to create batch, because we are not (and should not be) writing anything to database.

it will be just `return account.createKeyAtIndex(branch, index);`.